include/merlin/core_kernels/kernel_utils.cuh

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:///www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cooperative_groups.h>
+#include <cooperative_groups/reduce.h>
+#include <cuda_pipeline.h>
+#include <cuda/barrier>
+#include <mutex>
+#include <thread>
+#include <vector>
+#include "../types.cuh"
+#include "../utils.cuh"
+
+// if i % 2 == 0, select buffer 0, else buffer 1
+#define SAME_BUF(i) (((i)&0x01) ^ 0)
```

Comment:
Unused?

Reply:
No, it is used to select buffer in pipeline kernel. For example:
`V* v_src = sm_vector[SAME_BUF(i)][groupID];` in kernel lookup_kernel_with_io_pipeline_v2.

Reply:
Sorry, I didn't expand the `lookup_kernels.cuh`.
So here is a potential issue, the macro naming is too common that may dirty the end-users name scope, if no performance loss, can we change them to a `__forced_ inline__ __device__ func(..)` ?

Reply:
OK, I get